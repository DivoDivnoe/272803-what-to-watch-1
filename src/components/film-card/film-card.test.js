import React from 'react';
import {create} from 'react-test-renderer';
import FilmCard from './film-card.jsx';

const mock = {
  movie: {
    title: `Иван Васильевич меняет профессию`,
    image: `https://vokrug.tv/pic/news/d/d/8/4/dd84fb57fee799d831753daa2e0eb07e.jpg`,
    genre: `comedy`,
    preview: ``,
  },
  clickHandler: jest.fn(),
  handlePreview: jest.fn(),
  stopPreview: jest.fn(),
  isPlaying: false,
};

describe(`FilmCard component`, () => {
  it(`renders correctly`, () => {
    const {movie, clickHandler, handlePreview, stopPreview, isPlaying} = mock;

    const tree = create(
        <FilmCard
          movie={movie}
          clickHandler={clickHandler}
          handlePreview={handlePreview}
          stopPreview={stopPreview}
          isPlaying={isPlaying}
        />,
        {createNodeMock: (el) => {
          return el;
        }}).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
