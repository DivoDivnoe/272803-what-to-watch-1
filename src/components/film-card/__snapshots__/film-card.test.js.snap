// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard component renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={null}
>
  <video
    height="175"
    poster="https://vokrug.tv/pic/news/d/d/8/4/dd84fb57fee799d831753daa2e0eb07e.jpg"
    width="280"
  >
    Your browser does not support the video tag.
  </video>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Иван Васильевич меняет профессию
    </a>
  </h3>
</article>
`;
